### DISCUSSION TITLE
`flymake` not starting automatically with `eglot-ensure`

### DISCUSSION THREAD

**Discussion by ywwry66** [joaotavora/eglot#1404] [Q&A]
When starting a LSP server using `eglot-ensure`, the mode line shows "Flymake:?" and no `flymake` error/warning markups are displayed in the buffer. After a user command such as `C-n`, `flymake` starts to work correctly. This has nothing to do with the language server as it is the case with different languages, e.g., `C`, `R` and `latex`. Also, in the `*EGLOT events*` buffer, I can see the language server correctly sends `textDocument/publishDiagnostics` upon `eglot-ensure` being called. Furthermore, this `flymake` glitch only occurs with `eglot-ensure`, but not with `eglot`.

An MWE:
1. Start `Emacs` with `emacs -Q`
3. Open a buffer that is supported by `eglot`, e.g. `foo.c`
4. In the buffer evaluate `(eglot-ensure)`

Current behavior:
The mode line shows "Flymake:?" and no `flymake` error/warning markups are displayed in the buffer.

Expected behavior:
The mode line shows "Flymake[1 1]" and `flymake` error/warning markups are displayed correctly in the buffer.

**Comment by joaotavora** [comment]
> In the buffer evaluate (eglot-ensure)

This is not how you use `eglot-ensure`, so your recipe should be:

1. Start Emacs with emacs -Q knowing you have `clangd` installed
2. `(add-hook 'c-ts-mode-hook #'eglot-ensure)`
3. Visit foo.c

**Reply by joaotavora** [reply]
Must be some missing `force-mode-line-update` somewhere.  The problem goes away after any command, doesn't even have to be a buffer command, just anything that updates the mode-line.

**Reply by ywwry66** [reply]
> (add-hook 'c-ts-mode-hook #'eglot-ensure)

You are right this is my normal usage.
> Must be some missing `force-mode-line-update` somewhere.

I just tried to append `force-mode-line-update` to `eglot-ensure`, but with no luck. I am also not ensure whether a mode line update will simultaneously update the in-buffer `flymake` annotations.
